go/merkletree/merkle_verifier.go

```diff
@@ -47,15 +47,16 @@ func (m MerkleVerifier) VerifyInclusionProof(leafIndex, treeSize int64, proof []
 }
 
 // RootFromInclusionProof calculates the expected tree root given the proof and leaf.
+// leafIndex starts at 0. treeSize starts at 1.
 func (m MerkleVerifier) RootFromInclusionProof(leafIndex, treeSize int64, proof [][]byte, leaf []byte) ([]byte, error) {
-	if leafIndex > treeSize {
+	if leafIndex >= treeSize {
```

Comment:
Ah, I hadn't spotted that, good catch! Now, that also implies that leafIndex < treeSize, so therefore, treeSize has to be at least 1, it all falls together...